.github/workflows/rust.yml

```
@@ -54,3 +56,29 @@ jobs:
     - run: cargo hack --feature-powerset check --target ${{ matrix.sys.target }}
     - if: matrix.sys.test
       run: cargo hack --feature-powerset test --target ${{ matrix.sys.target }}
+
+  cargo-publish:
+    if: github.event_name == 'release'
+    needs: [complete]
+    runs-on: ubuntu-latest
+    steps:
+    - if: contains(needs.*.result, 'failure')
+      run: exit 1
+    - uses: actions/checkout@v3
+    - run: rustup update
+    - run: cargo publish
+    env:
+    - CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
```

It would be ideal if we only expose the secret to the publish command, and not everything else.
```suggestion
      env:
      - CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
```